### PR TITLE
refactor(client): Reduce fxa-js-client fetch boilerplate

### PR DESCRIPTION
**Pull request by shane-tomlinson**
In lib/fxa-client.js, add a `withClient` decorator function that
gets a fxa-js-client instance and passes it as the first parameter
to a function. This allows a lot of `this._getClient().then...` boilerplate
to be removed.

I didn't go all-in on ES6'ifying this, I wanted the PR to be semi-manageable to review.

@vbudhram - mind r?

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
Probably also easiest to review with the `w=1` query param

File: app/scripts/lib/fxa-client.js [hunk 1]
```diff
@@ -478,52 +443,43 @@ define(function (require, exports, module) {
      * @param {Relier} relier
      * @returns {Promise} resolves with new session information on success.
      */
-    changePassword: function (originalEmail, oldPassword, newPassword, sessionToken, sessionTokenContext, relier) {
+    changePassword: withClient((client, originalEmail, oldPassword, newPassword, sessionToken, sessionTokenContext, relier) => {
```
Comment (vbudhram) [inline app/scripts/lib/fxa-client.js]: nit: Add `client` to docs

Comment (shane-tomlinson) [inline app/scripts/lib/fxa-client.js]: I was unsure how to handle that, depends on the consumer of the API docs. A consumer of the API doesn't pass in `client` and doesn't need to know about it. A developer modifying the function does. I chose the POV of a consumer of the API.

**Comment by vbudhram** [comment]
@shane-tomlinson Took a little to get my head wrapped around it, but LGTM 👍, r+

**Pull request merged**